// firebase-tools/lib/crashlytics/events.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listEvents = void 0;
const logger_1 = require("../logger");
const error_1 = require("../error");
const utils_1 = require("./utils");
const filters_1 = require("./filters");
async function listEvents(appId, filter, pageSize = 1) {
    const requestProjectNumber = (0, utils_1.parseProjectNumber)(appId);
    try {
        const queryParams = (0, filters_1.filterToUrlSearchParams)(filter);
        queryParams.set("page_size", `${pageSize}`);
        logger_1.logger.debug(`[crashlytics] listEvents called with appId: ${appId}, filter: ${queryParams.toString()}, pageSize: ${pageSize}`);
        const response = await utils_1.CRASHLYTICS_API_CLIENT.request({
            method: "GET",
            headers: {
                "Content-Type": "application/json",
            },
            path: `/projects/${requestProjectNumber}/apps/${appId}/events`,
            queryParams: queryParams,
            timeout: utils_1.TIMEOUT,
        });
        return response.body;
    }
    catch (err) {
        throw new error_1.FirebaseError(`Failed to list events for app_id ${appId}.`, {
            original: (0, error_1.getError)(err),
        });
    }
}
exports.listEvents = listEvents;
